src/darwin/Framework/CHIP/EndpointDescription/MTRServerClusterDescription.h

```diff
@@ -0,0 +1,89 @@
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRAccessGrant.h>
+#import <Matter/MTRAttributeDescription.h>
+#import <Matter/MTRDefines.h>
+#import <Matter/MTRDeviceType.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * A representation of a server cluster implemented by an MTRDeviceController.
+ */
+MTR_NEWLY_AVAILABLE
+@interface MTRServerClusterDescription : NSObject <NSCopying>
+
+- (instancetype)init NS_UNAVAILABLE;
++ (instancetype)new NS_UNAVAILABLE;
+
+/**
+ * The provided clusterID must not be MTRClusterIDTypeDescriptorID; see
+ * initDescriptorCluster.
+ *
+ * Otherwise, it must be a valid cluster identifier.  That means:
+ *
+ * * In the range 0-0x7FFF for standard clusters.
+ * * In the range 0xVVVVFC00-0xVVVVFFFE for vendor-specific clusters, where VVVV
+ *   is the vendor identifier.
+ *
+ * The provided clusterRevision must be in the range 1-65535.
+ *
+ */
+- (nullable instancetype)initWithClusterID:(NSNumber *)clusterID clusterRevision:(NSNumber *)clusterRevision error:(NSError * __autoreleasing *)error;
+
+/**
+ * Create a cluster description for the descriptor cluster.  This will set
+ * clusterRevision to the current version implemented by Matter.framework.
+ */
+- (instancetype)initDescriptorCluster;
+
+@property (nonatomic, copy, readonly) NSNumber * clusterID;
+
+@property (nonatomic, copy, readonly) NSNumber * clusterRevision;
+
+/**
+ * A list of entities that are allowed to access this cluster instance.  This
+ * list is in addition to any endpoint-wide access grants that exist.
+ *
+ * Defaults to empty list, which means no additional access grants.
+ */
+@property (nonatomic, copy) NSArray<MTRAccessGrant *> * accessGrants;
+
+/**
+ * The list of attributes supported by the cluster and their initial values.
+ *
+ * If this cluster is the Descriptor cluster (id 0x001D), it must not define any
+ * values for DeviceTypeList, ServerList, ClientList; those values will be
+ * determined automatically.
+ *
+ * For all clusters, the global AttributeList, AcceptedCommandList,
+ * GeneratedCommandList attributes will be determined automatically and must not
+ * be included in the attributes array.
+ *
+ * For all clusters, the FeatureMap attribute will be assumed to be 0 unless
+ * otherwise specified and may be omitted from the attributes array.
+ *
+ * For all clusters, ClusterRevision will be determined automatically based on
```

Comment:
Maybe clusterRevision and featureMap should be treated the same? e.g. have properties for both that default to 0?

Reply:
0 is not a valid value for ClusterRevision.

More to the point, FeatureMap 0 is both valid per spec and very common for clusters.  But ClusterRevision is something that needs to be communicated to us by whoever is implementing the cluster, which in this case is the API consumer.